File: Extensions/Tessa.Extensions.Default.Client/AbTest/AbFilterViewDialogDescriptors.cs
using Tessa.Extensions.Default.Client.Views;
using Tessa.Views.Metadata.Criteria;

namespace Tessa.Extensions.Default.Client.AbTest
{
    /// <summary>
    /// Предоставляет объекты типа <see cref="FilterViewDialogDescriptor"/>.
    /// </summary>
    public static class AbFilterViewDialogDescriptors
    {
        /// <summary>
        /// Дескриптор, описывающий специальный диалог с параметрами фильтрации представления РМ AbTest/Автомобили.
        /// </summary>
        public static readonly FilterViewDialogDescriptor Cars =
            new("AbCarViewParameters",
            [
                new("CarName", "Parameters", "Name"),
                new("CarMaxSpeed", "Parameters", "MaxSpeed")
                {
                    CriteriaOperator = EqualsToCriteriaOperator.Instance
                },
                new("Driver", "Parameters", "DriverID")
                {
                    DisplayValueSectionName = "Parameters",
                    DisplayValueFieldName = "DriverName"
                },
                new("CarReleaseDateFrom", "Parameters", "ReleaseDateFrom"),
                new("CarReleaseDateTo", "Parameters", "ReleaseDateTo"),
            ]);
    }
}

File: WebClient SDK/src/default/abTest/abFilterViewDialogDescriptors.ts
import { ViewCriteriaOperators } from '@tessa/platform';

import { FilterViewDialogDescriptor } from '../views/filterViewDialogDescriptor';

/**
 * Предоставляет объекты типа {@link FilterViewDialogDescriptor}.
 */
export namespace AbFilterViewDialogDescriptors {
  /**
   * Дескриптор, описывающий специальный диалог с параметрами фильтрации представления РМ Администратор/Тестирование/Автомобили.
   */
  export const cars: FilterViewDialogDescriptor = {
    dialogName: 'AbCarViewParameters',
    parametersMapping: [
      { alias: 'CarName', valueSectionName: 'Parameters', valueFieldName: 'Name' },
      {
        alias: 'CarMaxSpeed',
        valueSectionName: 'Parameters',
        valueFieldName: 'MaxSpeed',
        criteriaOperator: ViewCriteriaOperators.EqualsTo
      },
      {
        alias: 'Driver',
        valueSectionName: 'Parameters',
        valueFieldName: 'DriverID',
        displayValueSectionName: 'Parameters',
        displayValueFieldName: 'DriverName'
      },
      {
        alias: 'CarReleaseDateFrom',
        valueSectionName: 'Parameters',
        valueFieldName: 'ReleaseDateFrom'
      },
      { alias: 'CarReleaseDateTo', valueSectionName: 'Parameters', valueFieldName: 'ReleaseDateTo' }
    ]
  };
}
